src/lib/features/metrics/client-metrics/metrics-service-v2.ts

```diff
@@ -103,6 +113,29 @@ export default class ClientMetricsServiceV2 {
         }
     }
 
+    async filterExistingToggleNames(toggleNames: string[]): Promise<string[]> {
+        if (this.flagResolver.isEnabled('filterExistingFlagNames')) {
+            try {
+                const validNames = await this.cachedFeatureNames();
+
+                const existingNames = toggleNames.filter((name) =>
+                    validNames.includes(name),
+                );
+                if (existingNames.length !== toggleNames.length) {
+                    this.logger.warn(
+                        `Filtered out ${toggleNames.length - existingNames.length} toggles with non-existing names`,
+                    );
+                }
+                return this.filterValidToggleNames(existingNames);
+            } catch (e) {
+                this.logger.error(e);
+                return this.filterValidToggleNames(toggleNames);
+            }
+        } else {
+            return this.filterValidToggleNames(toggleNames);
```

Comment:
Can be simplified to always return this.filterValidToggleNames(toggleNames); without the else and the return in the catch.

Reply:
yes, makes sense